File: JavaScript/Send_Emails/sendEmails.js
#! /usr/bin/env node

const inquirer = require('inquirer')
const auth = require('./auth')
let { authorization } = auth

//Package that enables sending of emails in Javascript
//npm install must be perfomed, in order to obtain all the modules
//In order to send the email successfully, the sender needs to enable "less secure apps" settings in the mail service used.

inquirer
.prompt([
        // Asks the user about their email
        {
                type: 'input',
                name: 'email',
                message: 'Enter your email:'
        },
])
.then(answers => {
        let email = answers.email
        // Asks the user the password of their email
        inquirer
        .prompt([
                {
                        type: 'input',
                        name: 'password',
                        message: 'Enter your password:'
                },
        ])
        .then(answers => {
                let password = answers.password
                // Asks the user about the receiver's email
                inquirer
                .prompt([
                        {
                                type: 'input',
                                name: 'to',
                                message: "Enter the receiver's email:"
                        },
                ])
                .then(answers => {
                        let to = answers.to
                        // Asks the user about the subject of the mail
                        inquirer
                        .prompt([
                                {
                                        type: 'input',
                                        name: 'subject',
                                        message: "Enter the subject:"
                                },
                        ])
                        .then(answers => {
                                let subject = answers.subject
                                // Asks the user to write the message that is needed to be sent to the receiver
                                inquirer
                                .prompt([
                                        {
                                                type: 'input',
                                                name: 'body',
                                                message: "Enter the message:"
                                        },
                                ])
                                .then(answers => {
                                        let message = answers.message

                                        let mailDetails = { 
                                                from: email, 
                                                to: to, 
                                                subject: subject, 
                                                text: message
                                        };

                                        authorization(email, password, mailDetails)
                                });
                        });
                });
        }); 
});

File: Python/Send_Emails/send_emails.py
#!/usr/bin/python

import smtplib
import getpass

# The smtp module (Simple Mail Transfer Protocol) enables sending emails in python
# The sender's email must be configured to less secure apps.
# This configuration can be made on visiting account information.
# Under the category security, less secure apps must turned on

# "Sender's Email-ID"
sender_email = input("Enter the sender's Email-ID : ")

# "Receiver's Email-ID"
receiver = input("Enter the receiver's Email-ID : ")

# "Sender's password"
password = getpass.getpass(prompt='Enter the sender's password : ')
                           
# "Subject of the Email"
subject = input("Enter the subject of the email : ")

# "Content of the email"
body_of_the_email = input("Enter the content of the email : ")

content = "Subject: {}\n\n{}".format(subject, body_of_the_email)

# Specifications of the Email

server = smtplib.SMTP("smtp.gmail.com", 587)

# Here the Gmail service is used, a different Email service can also be used
# The port 587, across which the email is sent

server.starttls()
server.login(sender_email, password)

# Login is authorised

print("Login success")

server.sendmail(sender_email, receiver, content)
print("Email sent to the receiver")

# Email is sent, prints success on sending the email
